sgl-kernel/csrc/cpu/decode.cpp

```diff
@@ -462,13 +463,17 @@ void decode_attention_kernel_impl(
     int64_t max_num_reqs,
     int64_t max_context_len,
     int64_t max_total_num_tokens) {
-  int64_t loc_val = loc[0];
-  for (int64_t i = 0; i < num_heads; i++) {
-    scalar_t* k_buffer_ptr = k_buffer + loc_val * k_strideN + i * k_strideH;
-    scalar_t* v_buffer_ptr = v_buffer + loc_val * v_strideN + i * v_strideH;
-    copy_stub<scalar_t>(k_buffer_ptr, key + i * head_size, head_size);
-    copy_stub<scalar_t>(v_buffer_ptr, value + i * head_size_v, head_size_v);
-  }
+  at::parallel_for(0, loc_size, GRAIN_SIZE / 64, [&](int64_t begin, int64_t end) {
```

Comment:
this should be `at::parallel_for(0, num_seqs * num_heads, 0, ...)`

num_seqs * num_heads should be big enough, just set grain_size to be 0 will be fine